client/src/pages/Profile.js

```diff
@@ -55,23 +57,55 @@ const useStyles = makeStyles((theme) => ({
       marginBottom: "20px",
     },
   },
+  // Loading
+  loadingSpinner: {
+    marginTop: "100px",
+  },
 }));
 
-function Profile() {
+function Profile({ match }) {
   const classes = useStyles();
 
+  // Get auth state
+  const { user, profile } = useContext(AuthStateContext);
+
+  const [profileDetails, setProfileDetails] = useState(null);
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      if (user && user._id === match.params.user_id) {
+        setProfileDetails(profile);
+      } else {
+        const result = await axios.get(`/api/profile/${match.params.user_id}`);
+        setProfileDetails(result.data);
+      }
+    };
+
+    fetchProfile();
+  }, [match.params.user_id, user, profile]);
+
   return (
     <Box className={classes.container + " " + classes.responsiveAlignment}>
-      <Paper className={classes.profileBreakpoints}>
-        <ProfileDetailCard />
-      </Paper>
-
-      {/* Add check to only display the request card if the user whose profile you are visiting is a sitter */}
-      <Paper
-        className={classes.requestBreakpoints + " " + classes.centerContent}
-      >
-        <ProfileRequestForm />
-      </Paper>
+      {profileDetails ? (
+        <>
+          <Paper className={classes.profileBreakpoints}>
+            <ProfileDetailCard profileDetails={profileDetails} />
+          </Paper>
+          {profileDetails.isSitter && user._id !== match.params.user_id && (
+            <Paper
+              className={
+                classes.requestBreakpoints + " " + classes.centerContent
+              }
+            >
+              <ProfileRequestForm />
```

Comment:
You should be able to easily integrate bookings by passing the userId as a prop to the Requests form

Reply:
Added.